node/router/src/helpers/sync.rs

```diff
@@ -95,9 +97,9 @@ pub struct Sync<N: Network> {
 
 impl<N: Network> Sync<N> {
     /// Initializes a new instance of the sync pool.
-    pub fn new(local_ip: SocketAddr) -> Self {
+    pub fn new(tcp: Tcp) -> Self {
```

Comment:
I can see this is a tough design request: Is there a way to preserve the original interface of merely passing in the `local_ip` since the `Sync` pool itself does not utilize `Tcp` functionalities?

Reply:
The fact that it needs to know the listening address already makes it dependent on the network stack. If it's desirable for it to be decoupled from it, it would be best if it could instead not use that address at all, delegating the related logic to a higher-level object.

Reply:
The only use of `local_ip` in `Sync` is for it to self identify itself in the `HashMap` of common ancestors. We could perhaps use a dummy IP for all cases of "self" then.